includes/CLI.php

```diff
@@ -159,29 +159,38 @@ public static function coupons( $args, $assoc_args ) {
 
 		$time_start = microtime( true );
 
-		$min = 5;
-		$max = 100;
-		if ( ! empty( $assoc_args['min'] ) ) {
-			$min = $assoc_args['min'];
-		}
-		if ( ! empty( $assoc_args['max'] ) ) {
-			$max = $assoc_args['max'];
-		}
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating coupons', $amount );
 
-		if ( $amount > 0 ) {
-			$progress = \WP_CLI\Utils\make_progress_bar( 'Generating coupons', $amount );
-			for ( $i = 1; $i <= $amount; $i++ ) {
-				Generator\Coupon::generate( true, $min, $max );
+		add_action(
+			'smoothgenerator_coupon_generated',
+			function () use ( $progress ) {
 				$progress->tick();
 			}
-			$progress->finish();
+		);
+
+		$remaining_amount = $amount;
+		$generated        = 0;
+
+		while ( $remaining_amount > 0 ) {
+			$batch = $remaining_amount > Generator\Coupon::MAX_BATCH_SIZE ? Generator\Coupon::MAX_BATCH_SIZE : $remaining_amount;
```

Comment:
Amounts to the same thing, but might be (marginally) more readable to just do:

```suggestion
			$batch = min( $remaining_amount, Generator\Coupon::MAX_BATCH_SIZE );
```

Optional, though, the outcome is the same and it's a bit subjective.

Reply:
It's a good call, I like that better. I guess I should port that to all the other commands as well...

Reply:
4e1ec2059150cf32a3a1c74e133362d1127ed5f3